File: android/practice_1/src/main/java/io/github/aidenkoog/style1/CustomApplication.java
package io.github.aidenkoog.style1;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.os.Build;
import android.view.inputmethod.InputMethodManager;

import androidx.multidex.MultiDex;

public class CustomApplication extends Application {
    public static InputMethodManager imm;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();
    }

    private boolean isDebuggable(Context context) {
        boolean debuggable = false;
        PackageManager pm = context.getPackageManager();
        try {
            ApplicationInfo appinfo = pm.getApplicationInfo(context.getPackageName(), 0);
            debuggable = (0 != (appinfo.flags & ApplicationInfo.FLAG_DEBUGGABLE));
        } catch (PackageManager.NameNotFoundException ignored) {
        }
        return debuggable;
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
    }

    private void setNotificationCannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            CharSequence name = "CHANNEL_NAME";
            String description = "CHANNEL_DESCRIPTION";
            int importance = NotificationManager.IMPORTANCE_DEFAULT;

            NotificationChannel channel
                    = new NotificationChannel(getPackageName(), name, importance);
            channel.setDescription(description);
            channel.setShowBadge(false);

            NotificationManager mNotificationManager
                    = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
            mNotificationManager.createNotificationChannel(channel);
        }
    }
}

File: android/in-review-projects/simple-news-list-app/app/src/main/java/io/github/aidenkoog/news_list/CustomApplication.kt
package io.github.aidenkoog.news_list

import android.app.Application
import io.github.aidenkoog.news_list.di.component.ApplicationComponent
import io.github.aidenkoog.news_list.di.component.DaggerApplicationComponent
import io.github.aidenkoog.news_list.di.module.ApplicationModule

class CustomApplication : Application() {

    lateinit var applicationComponent: ApplicationComponent

    override fun onCreate() {
        super.onCreate()
        injectDependencies()
    }

    private fun injectDependencies() {
        applicationComponent = DaggerApplicationComponent
            .builder()
            .applicationModule(ApplicationModule(this))
            .build()

        applicationComponent.inject(this)
    }
}